tests/performance/realtime.go

```diff
@@ -42,37 +39,39 @@ func byStartingTheVMI(vmi *v1.VirtualMachineInstance, virtClient kubecli.Kubevir
 	tests.WaitForSuccessfulVMIStart(vmi)
 }
 
-func byConfiguringTheVMIForRealtime(vmi *v1.VirtualMachineInstance, realtimeMask string) {
-	vmi.Spec.Domain.Resources.Limits = k8sv1.ResourceList{
-		k8sv1.ResourceMemory: memoryRequest,
-		k8sv1.ResourceCPU:    resource.MustParse("2"),
-	}
-	vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
-		k8sv1.ResourceMemory: memoryRequest,
-		k8sv1.ResourceCPU:    resource.MustParse("2"),
-	}
-	vmi.Spec.Domain.CPU = &v1.CPU{
-		Model:                 "host-passthrough",
-		DedicatedCPUPlacement: true,
-		Realtime:              &v1.Realtime{Mask: realtimeMask},
-		NUMA:                  &v1.NUMA{GuestMappingPassthrough: &v1.NUMAGuestMappingPassthrough{}},
-	}
-	vmi.Spec.Domain.Memory = &v1.Memory{
-		Hugepages: &v1.Hugepages{PageSize: "2Mi"},
-		Guest:     &memoryRequest,
+func withRelatimeConfiguration(realtimeMask string) libvmi.Option {
+	memoryRequest := resource.MustParse("512Mi")
+	return func(vmi *v1.VirtualMachineInstance) {
+		vmi.Spec.Domain.Resources.Limits = k8sv1.ResourceList{
+			k8sv1.ResourceMemory: memoryRequest,
+			k8sv1.ResourceCPU:    resource.MustParse("2"),
+		}
+		vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+			k8sv1.ResourceMemory: memoryRequest,
+			k8sv1.ResourceCPU:    resource.MustParse("2"),
+		}
+		vmi.Spec.Domain.CPU = &v1.CPU{
+			Model:                 "host-passthrough",
+			DedicatedCPUPlacement: true,
+			Realtime:              &v1.Realtime{Mask: realtimeMask},
+			NUMA:                  &v1.NUMA{GuestMappingPassthrough: &v1.NUMAGuestMappingPassthrough{}},
+		}
+		vmi.Spec.Domain.Memory = &v1.Memory{
+			Hugepages: &v1.Hugepages{PageSize: "2Mi"},
+			Guest:     &memoryRequest,
+		}
 	}
 }
 
 var _ = SIGDescribe("CPU latency tests for measuring realtime VMs performance", func() {
 
 	var (
-		vmi        *v1.VirtualMachineInstance
 		virtClient kubecli.KubevirtClient
-		err        error
```

Comment:
These changes are now mixed with the libvmi addition, but it would be easier to review and be confident that it is ok if you would have split them into separate commits, explaining why you did this.